File: src/main/java/io/github/kingg22/api/vacunas/panama/persistence/repository/FabricanteRepository.java
package io.github.kingg22.api.vacunas.panama.persistence.repository;

import io.github.kingg22.api.vacunas.panama.persistence.entity.Fabricante;
import java.util.Optional;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;

public interface FabricanteRepository extends JpaRepository<Fabricante, UUID> {

    Optional<Fabricante> findByLicencia(String licencia);

    Optional<Fabricante> findByUsuario_Id(UUID idUsuario);
}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/modules/fabricante/repository/FabricanteRepository.kt
package io.github.kingg22.api.vacunas.panama.modules.fabricante.repository

import io.github.kingg22.api.vacunas.panama.modules.fabricante.entity.Fabricante
import io.github.kingg22.api.vacunas.panama.util.withSession
import io.quarkus.hibernate.reactive.panache.kotlin.PanacheRepositoryBase
import io.smallrye.mutiny.coroutines.awaitSuspending
import jakarta.enterprise.context.ApplicationScoped
import java.util.UUID

@ApplicationScoped
class FabricanteRepository : PanacheRepositoryBase<Fabricante, UUID> {
    suspend fun findByIdOrNull(id: UUID): Fabricante? = withSession { findById(id).awaitSuspending() }

    suspend fun findByLicencia(licencia: String): Fabricante? = withSession {
        find("licencia = ?1", licencia).firstResult().awaitSuspending()
    }

    suspend fun findByUsuarioId(idUsuario: UUID): Fabricante? = withSession {
        find("usuario.id = ?1", idUsuario).firstResult().awaitSuspending()
    }
}
